From d094e64f90d5754c807c1984c1f706accf741936 Mon Sep 17 00:00:00 2001
From: James Dumay <jameswdumay@gmail.com>
Date: Thu, 11 Jun 2026 08:26:33 +1000
Subject: [PATCH 083/108] Allow DeepSeek2 NextN tensor mapping

---
 gguf-py/gguf/constants.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gguf-py/gguf/constants.py b/gguf-py/gguf/constants.py
index 463963f2..e0d7ad81 100644
--- a/gguf-py/gguf/constants.py
+++ b/gguf-py/gguf/constants.py
@@ -3060,6 +3060,12 @@ MODEL_TENSORS: dict[MODEL_ARCH, list[MODEL_TENSOR]] = {
         MODEL_TENSOR.FFN_DOWN_SHEXP,
         MODEL_TENSOR.FFN_UP_SHEXP,
         MODEL_TENSOR.FFN_EXP_PROBS_B,
+        MODEL_TENSOR.NEXTN_EH_PROJ,
+        MODEL_TENSOR.NEXTN_EMBED_TOKENS,
+        MODEL_TENSOR.NEXTN_ENORM,
+        MODEL_TENSOR.NEXTN_HNORM,
+        MODEL_TENSOR.NEXTN_SHARED_HEAD_HEAD,
+        MODEL_TENSOR.NEXTN_SHARED_HEAD_NORM,
     ],
     MODEL_ARCH.DEEPSEEK2OCR: [
         MODEL_TENSOR.TOKEN_EMBD,
-- 
2.50.1 (Apple Git-155)

